app/connectors/AddressLookupConnector.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import config.AddressLookupConfig
+import config.FrontendAppConfig
+import config.featureswitch.FeatureSwitching
+import models.incomeSourceDetails.viewmodels.httpparser.GetAddressLookupDetailsHttpParser.GetAddressLookupDetailsResponse
+import models.incomeSourceDetails.viewmodels.httpparser.PostAddressLookupHttpParser.PostAddressLookupResponse
+import play.api.libs.json.JsValue
+import play.api.libs.json.Json
+import play.api.mvc.{AnyContent, RequestHeader}
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpResponse}
+import controllers.routes
+import play.api.Logger
+import play.libs.Json
+
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class AddressLookupConnector @Inject()(val appConfig: FrontendAppConfig,
+                                       val addressConfig: AddressLookupConfig,
+                                       http: HttpClient)(implicit ec: ExecutionContext) extends FeatureSwitching {
+
+  val baseUrl: String = appConfig.addressLookupService
+
+  def addressLookupInitializeUrl: String = {
+    s"${baseUrl}/api/v2/init"
+  }
+
+  def getAddressDetailsUrl(id: String): String = {
+    s"${baseUrl}/api/v2/confirmed?id=$id"
+  }
+
+  lazy val individualContinueUrl: String = routes.AddBusinessAddressController.submit(None).url
+  lazy val agentContinueUrl: String = routes.AddBusinessAddressController.agentSubmit(None).url
+
+
+  //"http://localhost:9081/report-quarterly/income-and-expenses/view/income-sources/add/business-check-details"
+  def addressJson(continueUrl: String): JsValue = {
+    play.api.libs.json.Json.parse(
+      s"""
```

Comment:
I would suggest to generate this json using case classes + there are urls which need to be generate dynamically: i.e  http://localhost:9081/report-quarterly/income-and-expenses/view/sign-out etc